src/Chat/OpenAIChat.php

```diff
@@ -99,6 +99,23 @@ public function generateChat(array $messages): string
         return $answer->choices[0]->message->content ?? '';
     }
 
+    public function generateChatOrReturnFunctionCalled(array $messages): string|FunctionInfo
+    {
+        $openAiArgs = $this->getOpenAiArgs($messages);
+        $answer = $this->client->chat()->create($openAiArgs);
+        $toolsToCall = $this->getToolsToCall($answer);
+
+        foreach ($toolsToCall as $toolToCall) {
+            $this->lastFunctionCalled = $toolToCall;
+        }
+
+        if ($this->lastFunctionCalled instanceof FunctionInfo) {
```

Comment:
Hey @samuelgjekic,
I think it is great to add this possibility.

what if the previous answer was a function and now it is a text but the lastFunctionCalled is still the FunctionInfo from the previous call?

Reply:
Hmm i did not think about this, my bad.. Anyways, does this not apply to the generateTextOrReturnFunction aswell? As they are the same. 

I added a line that sets the lastFunction to null in the start of the method, is this good, or do you prefer that i check if **$toolstocall** is not null instead? If so dont merge yet and i will do it that way instead.